cada/commands.py

```diff
@@ -223,9 +225,21 @@ def anon():
         writer = csv.writer(csvfile)
         # Generate header
         writer.writerow(csv.ANON_HEADER)
-
+        match_regex = r'(M\.|Mme\.|Monsieur|Madame|Docteur|Mademoiselle)\s+([\w\']+)'
         for idx, advice in enumerate(candidates, 1):
-            writer.writerow(csv.to_anon_row(advice))
+            replace = []
+            for possible_location in [advice.subject, advice.content]:
+                matches = re.findall(match_regex, possible_location)
+                if not matches:
+                    continue
+                good_matches = [m[1] for m in matches if len(m) > 1 and len(m[1]) > 3 and m[1][0].isupper()]
```

Comment:
With this type of queries on MongoDB, can you recover within the `candidate` variable  the actual match of these regexes ? If not, I still need to re-read both (advice and subject) looking for the actual match.

Reply:
If I understood correctly, we want to get those things that are not XXX (`name_regex1`) and then filter them with `name_regex2`. I tried doing this quickly, but it does not seem to work:

```
    candidates = Advice.objects(__raw__={

        '$and': [{
                "$or": [{
                    'subject': {
                        '$regex': name_regex1,
                        '$options': 'imx',
                    }
                },
                    {
                    'content': {
                        '$regex': name_regex1,
                        '$options': 'imx',
                    }}
                ]},

                {
                '$or': [{
                    'subject': {
                        '$regex': match_regex,
                        '$options': 'mx',
                    }
                },
                    {
                    'content': {
                        '$regex': match_regex,
                        '$options': 'mx',
                        }
                    }
                ]}
            ]})
```

I think that a single regex like this `match_regex = r'(%s)\s+([A-WYZ][^X\.\-][\w]+)' % PREFIXES` would do the job. The dot and dash to avoid _P.a_ or _Z-M_ (already anonymized text).

What do you think ?

Reply:
that seems ok to me